src/clients/Client.cc

```diff
@@ -847,18 +849,17 @@ void Client::handleAdaptedBodyProducerAborted()
     if (abortOnBadEntry("entry went bad while waiting for the now-aborted adapted body"))
         return;
 
+    adaptedReplyAborted = true;
     Must(adaptedBodySource != nullptr);
     if (!adaptedBodySource->exhausted()) {
         debugs(11,5, "waiting to consume the remainder of the aborted adapted body");
         return; // resumeBodyStorage() should eventually consume the rest
     }
 
-    stopConsumingFrom(adaptedBodySource);
```

Comment:
Can we restore this call? We already set adaptedReplyAborted. We know there is nothing in that BodyPipe buffer. Why keep adaptedBodySource? checkAdaptationCompletion() can assume that nil adaptedBodySource means that either there was no body at all or the entire body (whole or aborted) was consumed -- in either case, it will not wait for resumeBodyStorage().

In general, we want to cleanup ASAP.

Reply:
I moved this method to checkAdaptationCompletion(). In this context, if we get to checkAdaptationCompletion(), it will definitely call stopConsumingFrom(). If we stopConsumingFrom(), we would need to adjust the general checkAdaptationCompletion() to account for this precondition, which would complicate that method.